web/tests/reactions.test.js

```diff
@@ -369,6 +369,24 @@ test("sending", ({override, override_rewire}) => {
     });
 });
 
+test("prevent_simultaneous_requests_updating_reaction", ({override, override_rewire}) => {
+    const message = {...sample_message};
+    override(message_store, "get", (message_id) => {
+        assert.equal(message_id, message.id);
+        return message;
+    });
+    override_rewire(reactions, "add_reaction", () => {});
+    const stub = make_stub();
+    channel.post = stub.f;
+
+    // Verify that two requests to add the same reaction in a row only
+    // result in a single request to the server.
+    reactions.toggle_emoji_reaction(message.id, "cow");
+    reactions.toggle_emoji_reaction(message.id, "cow");
```

Comment:
The previous version of this test didn't actually verify the logic. 

+test("prevent_simultaneous_requests_updating_reaction", ({override}) => {
+    const message = {...sample_message};
+    override(message_store, "get", (message_id) => {
+        assert.equal(message_id, message.id);
+        return message;
+    });
+    override(
+        reactions.view,
+        "loading_reaction_ids",
+        new Set([`unicode_emoji,1f47d,${message.id}`]),
+    );
+    const stub = make_stub();
+    channel.post = stub.f;
+
+    reactions.toggle_emoji_reaction(message.id, "alien");
+
+    assert.equal(stub.num_calls, 0);
+});

This test didn't actually verify the logic, both because the `loading_reaction_ids` mock results in not verifying the logic of adding items to the new `Set`, and second because the `alien` emoji had been added in previous tests in the file.

This can be rewritten much more simply to just try to toggle the same emoji reaction twice without running code to confirm a result from the server and confirming the second doesn't result in a server call; I've done that, and checked that with the old code, we'd have gotten two calls to the stub.